BWEnv/src/controller.cc

```diff
@@ -434,14 +434,68 @@ int8_t Controller::handleCommand(int command, const std::vector<int>& args,
       draw_cmds_.push_back({cmd, str});
       return CommandStatus::SUCCESS;
     }
-    case Commands::COMMAND_USER:
+    case Commands::COMMAND_USER: {
       auto type = args[0];
       auto second = args.begin() + 1;
       auto last = args.end();
       std::vector<int> user_args(second, last);
 
       return handleUserCommand(type, user_args);
     }
+    case Commands::COMMAND_OPENBW: {
+      // includes type
+      static std::unordered_map<int, int> obw_argcount = {
+        {OBWCommands::KILL_UNIT, 3}, {OBWCommands::SPAWN_UNIT, 5},
+      };
+      if (!check_args(obw_argcount[command])) return status;
+      auto type = args[0];
+      auto second = args.begin() + 1;
+      auto last = args.end();
+      std::vector<int> user_args(second, last);
+
+      return handleOpenBWCommand(type, user_args);
+    }
+    }
+    Utils::bwlog(output_log, "Invalid command: %d", command);
+    return CommandStatus::UNKNOWN_COMMAND;
+  }
+}
+
+int8_t Controller::handleOpenBWCommand(int, command, const std::vector<int>& args)
+{
+  #ifndef OPENBW_BWAPI
+  status = CommandStatus::OPENBW_NOT_IN_USE;
+  return status;
+  #endif
```

Comment:
You might want an #else here and #endif at the end of the function, otherwise you'd get compile errors due to functions createUnit and killUnit not existing.